File: web-view/bootstrap/src/support/GenericLocalization.js
import "webl10n/l10n";

const webL10n = document.webL10n;

export class GenericLocalization {
  constructor(lang) {
    this._lang = lang;
    this._ready = new Promise(resolve => {
      webL10n.setLanguage(lang, () => {
        resolve(webL10n);
      });
    });
  }

  async getLanguage() {
    const l10n = await this._ready;
    return l10n.getLanguage();
  }

  async getDirection() {
    const l10n = await this._ready;
    return l10n.getDirection();
  }

  async get(property, args, fallback) {
    const l10n = await this._ready;
    return l10n.get(property, args, fallback);
  }

  async translate(element) {
    const l10n = await this._ready;
    return l10n.translate(element);
  }
}

File: web-view/pdfjs/src/main/kotlin/pdfjs/web/GenericLocalization.kt
// @file:JsModule("pdfjs-dist/lib/web/genericl10n")
package pdfjs.web

import kotlin.js.Promise

// @JsName("GenericL10n")
@OptIn(ExperimentalJsExport::class)
@JsExport
class GenericLocalization(private val language: String) {
  fun getLanguage(): Promise<String> {
    return Promise.resolve(language)
  }

  fun getDirection(): Promise<dynamic> {
    return Promise.resolve("ltr")
  }

  fun get(key: String, arguments: dynamic, fallback: dynamic): Promise<String> {
    return Promise.resolve("template string")
  }

  fun translate(element: dynamic): Promise<dynamic> {
    return Promise.resolve("template string##")
  }
}
